programs/summerofcode/2025.md

```diff
@@ -94,6 +94,38 @@ You can find the project ideas from previous year [here](./2024.md).
   - Heipa (@He1pa, he1pa404@gmail.com)
 - Upstream Issue (URL): https://github.com/kcl-lang/kpm/issues/598
 
+#### KubeBuilder
+
+##### Automating Operator Maintenance: Driving Better Results with Less Overhead
+
+- Description:  
+Code-generation tools like Kubebuilder and Operator-SDK have transformed cloud-native application development by providing scalable, community-driven frameworks. These tools simplify complexity, accelerate results, and enable developers to create tailored solutions while avoiding common pitfalls, laying a strong foundation for innovation.  
+However, as these tools evolve with ecosystem changes and new features, projects risk becoming outdated. Manual updates are time-consuming, error-prone, and make it challenging to maintain security, adopt advancements, and stay aligned with modern standards.  
+This project introduces an automated solution for Kubebuilder, with potential applications for similar tools or those built on its foundation. By streamlining maintenance, projects remain aligned with modern standards, improve security, and adopt the latest advancements. It fosters growth and innovation across the ecosystem, letting developers focus on what matters most: building great solutions.  
+Note that the initial idea is to solve this with **3-way Git merges**. However, users will face conflicts, and in the first phase, we want to study whether AI could help resolve these conflicts in a future phase to achieve this goal.
+
+- Expected Outcome
+  - Conduct research on [3-Way Merge & Advanced Merge Options in Git](https://youtu.be/3Kb9glVJBJM?si=a0g3QMSrW_CMl1L9).
+  - **Conduct research on how AI could help resolve conflicts**. If open-source solutions are available and align with the proposal, include them for consideration in a second phase.
+  - Develop a **Proof of Concept (POC)** implementing a **GitHub Action** that automatically creates a **Pull Request (PR)** in a mock repository, demonstrating the feasibility of the proposed solution.
+  - Successfully complete the proposal for [PEP](https://github.com/kubernetes-sigs/kubebuilder/pull/4302).
+  - Introduce a new [Kubebuilder Plugin](https://book.kubebuilder.io/plugins/plugins) that scaffolds the **GitHub Action** based on the POC. This plugin will be released as an **alpha feature**, allowing users to opt-in for automated updates. The initial solution does **not need to have AI**, but AI integration could be a future enhancement if feasible.
+
+- Recommended Skills
+  - Golang
+  - GitHub Actions
+  - Software Automation
+  - CI/CD
+  - Git
+  - IA
+
+- Expected project size: **Large** (~350 hour projects)
+
+- Mentor(s)
+  - Camila Macedo (@camilamacedo86, camilamacedo86@gmail.com) - Primary
+  - Varsha Prasad (@varshaprasad96, varshaprasad1507@gmail.com)
```

Comment:
```suggestion
  - Varsha Prasad (@varshaprasad96, varshaprasad1507@gmail.com)
  - TianYi(Tony) (@Kavinjsir)
```

@nate-double-u also, just to share, we both are admins of the project: 
https://github.com/kubernetes-sigs/kubebuilder/blob/master/OWNERS_ALIASES#L6-L8

Also, @Kavinjsir was a mentee in the past, and he just confirmed that he wants to join us as a mentor this year. That’s definitely proof of the success of this program! 😊

I hope we’re still in time to add him.

Reply:
By all means add them, it's great seeing a mentee become a mentor!

We can still make changes, tomorrow's deadline is just the org application.

We ping sig-contribex because they head up the mentoring program over at K8s. This is just a function of kubernetes sub project admins not being listed as maintainers on the overall Kubernetes project, and it's usually maintainers that need to approve projects. Kubernetes is just set up different :)

KubeBuilder has participated a bunch of times, I don't think it'll be an issue.